pip_audit/_dependency_source/requirement.py

```diff
@@ -313,12 +318,21 @@ def _collect_preresolved_deps(
             if req.marker is not None and not req.marker.evaluate():
                 continue  # pragma: no cover
 
-            # This means we have a duplicate requirement for the same package
-            if req.name in req_names:
+            duplicate_req_specifier = req_specifiers.get(req.name)
+
+            if not duplicate_req_specifier:
+                req_specifiers[req.name] = req.specifier
+
+            # We have a duplicate requirement for the same package
+            # but different specifiers, meaning a badly resolved requirements.txt
+            elif duplicate_req_specifier != req.specifier:
                 raise RequirementSourceError(
                     f"package {req.name} has duplicate requirements: {str(req)}"
                 )
-            req_names.add(req.name)
+            else:
+                # We have a duplicate requirement for the same package and the specifier matches
+                # As they would return the same result from the audit, there no need to yield it a second time.
+                continue  # pragma: no cover
```

Comment:
Sorry, but I think we need to do better than skipping coverage on this -- it's a core part of the change, so we can't just skip it. 

This PR has been open for a long time (that's my fault), so I'll take over here and try to rewrite it into a shape that `coverage.py` understands.

Reply:
Does it really worth that much effort ? Especially as coverage does not complain on py3.10+, and technically the problem come from coverage itself 🤔

Reply:
Hmm, good point. I'll update the comment to say we can drop the ignore after we're 3.10+ and after that this LGTM.